File: 은지/LV1/비밀지도.py
def two(x):
    if x < 2:
        return x
    else:
        return two(x//2) * 10 + x % 2

def solution(n, arr1, arr2):
    answer = []
    # 지도 초기화
    Map = [[0]*n for _ in range(n)]
    for i in range(n):
        # 10진수 => 2진수
        a1 = str(two(arr1[i]))
        a2 = str(two(arr2[i]))
        # 벽 입력
        for j in range(len(a1)-1, -1, -1):
            Map[i][j+n-len(a1)] += int(a1[j])
        for j in range(len(a2)-1, -1, -1):
            Map[i][j+n-len(a2)] += int(a2[j])
    # 지도 출력
    for i in range(n):
        result = ""
        for j in range(n):
            if Map[i][j]:
                result += "#"
            else:
                result += " "
        answer.append(result)
    return answer

File: 은지/LV1/비밀지도.js
function solution(n, arr1, arr2) {
  var answer = [];
  // 지도 초기화
  var map = new Array(n)
  for (let i=0; i<n; i++) {
    map[i] = new Array(n)
  }
  for (let i=0; i<n; i++) {
    // 10진수 => 2진수
    var a1 = arr1[i].toString(2)
    var a2 = arr2[i].toString(2)
    // 벽 입력
    for (let j=a1.length-1; j>=0; j--) {
      map[i][j+n-a1.length] = Number(a1[j])   
    }
    for (let j=a2.length-1; j>=0; j--) {
      if (map[i][j+n-a2.length] >= 0) {
        map[i][j+n-a2.length] += Number(a2[j])
      } else {
      map[i][j+n-a2.length] = Number(a2[j])
      }      
    }
  }
  // 지도 출력
  for (let i=0; i<n; i++) {
    var result = ""
    for (let j=0; j<n; j++) {
      if (map[i][j]) {
        result += "#"
      } else {
        result += " "
      }
    }
    answer.push(result)
  }
  return answer;
}